### PR TITLE
Add rich rule description fields to SQ Client 

### LINKED ISSUE #3761: Add rule descriptions to rules service

### PR DESCRIPTION
**Pull request by ugras-ergun-sonarsource**
fixes #3761

### CODE REVIEW

File: src/SonarQube.Client/Api/V5_50/GetRulesRequest.cs [hunk 1]
```diff
@@ -126,5 +134,36 @@ private sealed class ParameterResponse
...
+            [JsonProperty("key")]
+            public string Key { get; set; }
+
+            [JsonProperty("content")]
+            public string HtmlContent { get; set; }
+
+            [JsonProperty("context")]
+            public ContextResponse Context { get; set; }
+
+            internal SonarQubeDescriptionSection ToSonarQubeDescriptionSection()
```
Comment (georgii-borovinskikh-sonarsource) [inline src/SonarQube.Client/Api/V5_50/GetRulesRequest.cs:149]: what's the purpose of this abstraction layer?

File: src/SonarQube.Client/Models/SonarQubeDescriptionSection.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarQube.Client.Models
+{
+    public class SonarQubeDescriptionSection
```
Comment (georgii-borovinskikh-sonarsource) [inline src/SonarQube.Client/Models/SonarQubeDescriptionSection.cs:23]: Don't we already have DescriptionSections?

**Review by georgii-borovinskikh-sonarsource — CHANGES_REQUESTED** [review]
I'm a bit confused about duplicate classes for DescriptionSection and Context

File: src/SonarQube.Client/Models/SonarQubeDescriptionSection.cs (hunk 2 above)
Comment (ugras-ergun-sonarsource) [reply src/SonarQube.Client/Models/SonarQubeDescriptionSection.cs:23]: One of them is for parsing the response from the server which is a private class and this one is for the response object. This is for the logical separation of those endpoints.

Comment (georgii-borovinskikh-sonarsource) [reply src/SonarQube.Client/Models/SonarQubeDescriptionSection.cs:23]: But you also have the one that's used in IRuleDescription, which means you have 3 identical classes, no?

Comment (georgii-borovinskikh-sonarsource) [reply src/SonarQube.Client/Models/SonarQubeDescriptionSection.cs:23]: IRuleInfo*

**Review by georgii-borovinskikh-sonarsource — APPROVED** [review]

**Pull request merged**